DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -450,17 +450,32 @@ private async Task<DiscordApplicationCommandOption> ToApplicationParameterAsync(
         }
 
         Dictionary<CommandParameter, object?> parsedArguments = [];
+        CommandParameter? autoCompleteParameter = null;
+        DiscordInteractionDataOption? autoCompleteOption = null;
         try
         {
             // Parse until we find the parameter that the user is currently typing
-            while (converterContext.NextParameter() && !converterContext.Options.ElementAt(converterContext.ParameterIndex).Focused)
+            while (converterContext.NextParameter())
             {
+                DiscordInteractionDataOption? option = converterContext.Options.FirstOrDefault(x => x.Name.Equals(converterContext.Parameter.Name));
+                if (option is not null && option.Focused)
+                {
+                    autoCompleteParameter = converterContext.Parameter;
+                    autoCompleteOption = option;
+                    break;
+                }
+
                 IOptional optional = await this.ConverterDelegates[this.GetConverterFriendlyBaseType(converterContext.Parameter.Type)](converterContext, eventArgs);
                 parsedArguments.Add(converterContext.Parameter, optional.HasValue
                     ? optional.RawValue
                     : converterContext.Parameter.DefaultValue
                 );
             }
+
+            if (autoCompleteParameter is null || autoCompleteOption is null)
+            {
+                throw new InvalidOperationException("Cannot find the parameter that the user is currently typing.");
```

Comment:
In what scenario's will this happen? If it's expected behavior then it should return `null` in my opinion.

Reply:
I don't really know, i just put that in case but it should not happen. I can return null instead if you think it's better.

Reply:
Ok, i've faced a case when it appear if the user is too fast like spamming tab so I've made the change to return null.

![aGWQkICL43](https://github.com/DSharpPlus/DSharpPlus/assets/32265676/c195eeb2-7610-4ed8-9694-4e028a6b2999)
![Screenshot 2024-04-28 232033](https://github.com/DSharpPlus/DSharpPlus/assets/32265676/b7da9a21-af85-4c78-8094-067bbbe39264)

Reply:
Update, i find the issue... pfiou not simple to debug

Reply:
Ping me in #lib-development so everyone can work together on this issue